singer_sdk/connectors/sql.py

```diff
@@ -441,21 +468,34 @@ def discover_catalog_entry(
 
         # Initialize columns list
         table_schema = th.PropertiesList()
-        for column_def in inspected.get_columns(table_name, schema=schema_name):
-            column_name = column_def["name"]
-            is_nullable = column_def.get("nullable", False)
-            jsonschema_type: dict = self.to_jsonschema_type(
-                t.cast(sqlalchemy.types.TypeEngine, column_def["type"]),
-            )
-            table_schema.append(
-                th.Property(
-                    name=column_name,
-                    wrapped=th.CustomType(jsonschema_type),
-                    required=not is_nullable,
-                ),
-            )
+        with warnings.catch_warnings(record=True) as inspection_warnings:
+            for column_def in inspected.get_columns(table_name, schema=schema_name):
+                column_name = column_def["name"]
+                is_nullable = column_def.get("nullable", False)
+                jsonschema_type: dict = self.to_jsonschema_type(
+                    t.cast(sqlalchemy.types.TypeEngine, column_def["type"]),
+                )
+                table_schema.append(
+                    th.Property(
+                        name=column_name,
+                        wrapped=th.CustomType(jsonschema_type),
+                        required=not is_nullable,
+                    ),
+                )
+        if len(inspection_warnings) > 0:
+            for line in inspection_warnings:
+                expanded_msg: str = (
+                    f"Discovery warning: {line.message} in '{unique_stream_id}'"
+                )
+                self.logger.info(expanded_msg)
         schema = table_schema.to_dict()
 
+        sql_datatypes = {}
+        for column_def in inspected.get_columns(table_name, schema=schema_name):
+            sql_datatypes[
+                str(column_def["name"])
+            ] = self.discover_catalog_entry_sql_datatype(column_def["type"])
```

Comment:
Would you feel comfortable adding tests for this? A good place might be in `tests/core/test_connector_sql.py` since this is a connector method. I think it could even be parametrized with many examples of column types.